### PR TITLE
Upgrade Angular UI Router to v1.0.7

### PR DESCRIPTION
**Pull request by jaredevantabor**
##### SUMMARY
Relates to https://github.com/ansible/awx/issues/327 for updating from a beta version of angular-ui-router to the major version. This PR includes the changes to packaging for upgrading ui-router as well as fixes to make Tower functional (for the most part). 

##### ISSUE TYPE
 - Feature/Tech Debt Pull Request 

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
awx: 1.0.1.5
```

##### ADDITIONAL INFORMATION
To run the AWX UI with these new changes you'll likely need to do a `make clean-ui` and then `npm install`. Let me know if this doesn't work for you. 

I've found that the [migration guide](https://ui-router.github.io/guide/ng1/migrate-to-1_0) is a good resource. The major breaking changes that I address in this PR are 

- Changes to lazyLoad (we lazy load a lot of states via the state-generator). Now, when lazy loading states, the top most state name must end with a `.**`, so I added that in the appropriate places.
- Removing (the deprecated) `stateChangeStart`/`stateChangeSuccess` events, and replacing them with `$transitions` 
- Changes to where resolves and controllers can be defined within a state definition object 

Known issues that I'm still working: 

- Issues with search queries not showing up. This is pretty big, and I'd like to fix this before merging. 
- Issues with scrolling to the top of the view when clicking to edit an item from a very long list. 
- The tabTitle that displays the `AWX | <state name>` appears to be broken. It's getting set correctly on scope but the tab title isn't reflecting this. Not even sure if this is directly tied to the ui-router upgrade 

Other other: in the migration doc it mentions that $stateParams is going to be deprecated, so they recommend transitioning to using `$transitions.params()` instead. We use $stateParams a lot. Like a lot a lot. I'm not going to even try and replace all these instances, but I'll try and start testing that migration to see how painful it is. It seems like something we'll need to do over time. For now, $stateParams still works, so I think we're ok... for now

image

### CODE REVIEW

**Review by jlmitch5 — APPROVED** [review]
Looks geat!

I'd say, you could squash the commits before merging.

**Comment by gconsidine** [comment]
Nice, @jaredevantabor. The couple things I did notice are:

* `npm run unit` has 3 failing tests (I see a reference to `$stateChangeSuccess` so maybe that has something to do with it).
* `npm test e2e -- --env=debug` has 3 failed tests as well (in the smoke test suite).

Are you able to verify these tests pass on your side? The unit and e2e tests along with ESLint aren't yet integrated into the checks on open PRs -- we'll have to verify this stuff locally before merging anything in for now.

**Review by gconsidine — CHANGES_REQUESTED** [review]
I'll switch this over to "Approve" when the tests pass. Great work, tho. There were a lot of moving parts.

File: awx/ui/client/src/instance-groups/instance-groups.route.js
Comment (gconsidine) [inline awx/ui/client/src/instance-groups/instance-groups.route.js]: I see `dynamic` (correct spelling) changed to `dyanmic` (incorrect spelling) in several places -- does this mean that leaving `dynamic` unset/undefined leads to the correct default behavior from UI router? If so, we should remove these lines.

File: awx/ui/client/src/instance-groups/main.js
Comment (gconsidine) [inline awx/ui/client/src/instance-groups/main.js:54]: Is the `name.**` part documented somewhere in the ui-router docs? I can't seem to find information on that syntax (and others I know we've used).

Comment (jaredevantabor) [reply awx/ui/client/src/instance-groups/main.js:54]: @gconsidine https://ui-router.github.io/ng1/docs/latest/interfaces/ng1.ng1statedeclaration.html#lazyload 

```
A future state can act as a placeholder for a single state, or for an entire module of states and
substates. A future state should have:

A name which ends in .**. A future state's name property acts as a wildcard Glob. It matches
any state name that starts with the name (including child states that are not yet loaded).
```

**Comment by jaredevantabor** [comment]
With the help of @marshmalien I was able to get the outstanding bugs fixed before merging this PR

- Issues with search queries not showing up. This is pretty big, and I'd like to fix this before merging. ✅ 
- Issues with scrolling to the top of the view when clicking to edit an item from a very long list. ✅ 
- The tabTitle that displays the AWX | <state name> appears to be broken. It's getting set correctly on scope but the tab title isn't reflecting this. Not even sure if this is directly tied to the ui-router upgrade ✅

**Review by gconsidine — APPROVED** [review]

**Pull request merged**
